src/Models/BaseElement.php

```diff
@@ -528,6 +536,85 @@ public function getContentForSearchIndex(): string
         return $content;
     }
 
+    /**
+     * Provides content for CMS search
+     */
+    public function getContentForCmsSearch(): string
+    {
+        $textualDatabaseFields = $this->getTextualDatabaseFields();
+        $excludedFields = $this->getFieldsExcludedFromCmsSearch();
+        $contents = [];
+        foreach ($this->config()->get('db') as $fieldName => $fieldType) {
+            if (!in_array($fieldType, $textualDatabaseFields) || in_array($fieldName, $excludedFields)) {
+                continue;
+            }
+            $contents[] = $this->$fieldName;
+        }
+        // Use |#| to delimit different fields rather than space so that you don't
+        // accidentally join results of two columns that are next to each other in a table
+        $delimiter = '|#|';
+        $content = implode($delimiter, $contents);
+
+        // Strips tags and be sure there's a space between words.
+        $content = trim(strip_tags(str_replace('<', ' <', $content)));
+
+        // Allow projects to update content of third-party elements.
+        $content .= $delimiter;
+        $this->extend('updateContentForCmsSearch', $content);
+
+        // Remove delimiters that aren't required
+        $content = $this->removeExtraDelimiters($content, $delimiter);
+
+        return $content;
+    }
+    /**
+     * Get fields that have a Varchar or Text like type in the database
+     */
+    private function getTextualDatabaseFields(): array
+    {
+        $textualDatabaseFields = singleton(DataObjectSchema::class)->databaseFields($this);
+        foreach ($textualDatabaseFields as $fieldName => $databaseFieldType) {
+            $lcType = strtolower($databaseFieldType);
+            if (!str_contains($lcType, 'varchar') && !str_contains($lcType, 'text')) {
+                unset($textualDatabaseFields[$fieldName]);
+            }
+        }
+        return $textualDatabaseFields;
```

Comment:
I don't want to use `instanceof DBString` because that includes Enum's which are really just fancy integers rather than unstructured text content.  Projects could also do something weird with a subclass of DBString, so seems sensible to me to just check the database type

I'm not sure what you're trying to do with `strtolower(strtok($databaseFieldType ?? '', '('););`? It just makes the code harder to read. There's no differences doing a str_contains in `Varchar(255)` vs 'Varchar`>